nemo_aligner/utils/distributed.py

```diff
@@ -320,3 +324,37 @@ def sync_and_consume_over_stored_time(self, name=""):
         yield from output_list
 
         del self.stored_results[name]
+
+
+@dataclass
+class Timer:
+    """Timer to tell us when the time limit has reached
+    """
+
+    duration: Optional[str]
+
+    def __post_init__(self):
```

Comment:
Just curious why make it a dataclass with `__post_init__()` instead of a regular class with a regular `__init__()`?

Reply:
No reason other than being too lazy to write the `__repr__` etc for the class